src/paths/Path.zig

```diff
@@ -963,6 +963,53 @@ pub fn Path(comptime opts: Options) type {
     };
 }
 
+/// Run the function only on specific platforms, or always.
+pub const ExecutionOption = enum {
+    only_on_windows,
+    only_on_posix,
+    always,
+};
+
+/// Given a path, normalize it to use POSIX-style separators. Mutates the given string.
+///
+/// You may pass the `.only_on_windows` option to skip normalization on non-Windows platforms,
+/// saving a couple of cycles.
+pub fn normalizeSeparatorsMut(
+    input_path: []u8,
+    comptime opts: struct {
+        run_on: ExecutionOption = .always,
+    },
+) void {
+    if (opts.run_on == .only_on_windows and !bun.Environment.isWindows) {
+        return;
+    }
```

Comment:
I don't know how we feel about this -- the idea was to have a declarative way to disable the execution of this function so that the user could guard features which are relevant to the windows execution environment with this, and non-windows platforms would be none-the-wiser to this.

So from the API caller side, it looks like
```zig
normalizeSeparatorsMut(input, .{ .run_on = .only_on_windows });
for (bun.string.split(input)) |segment| {
}
```

You don't need `if-else`s this way, but the downside is maybe the code is less readable. Idk, I was trying to be clever and I'm no longer so sure this is that clever.